x-pack/plugins/cloud/public/components/chat/iframe.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useRef, useState, CSSProperties } from 'react';
+import { css } from '@emotion/react';
+import { useChat } from '../../services';
+
+type UseChatType =
+  | { enabled: false }
+  | {
+      enabled: true;
+      src: string;
+      ref: React.MutableRefObject<HTMLIFrameElement | null>;
+      style: CSSProperties;
+    };
+
+const MESSAGE_READY = 'driftIframeReady';
+const MESSAGE_RESIZE = 'driftIframeResize';
+const MESSAGE_SET_CONTEXT = 'driftSetContext';
+
+const iframeStyle = css`
+  position: fixed;
+  botton: 30px;
+  right: 30px;
+  display: block;
+`;
+
+// We're sending a lot of information to the frame, so this method puts together the specific
+// properties, 1/ to avoid leaking too much, and 2/ to enumerate precisely what we're sending.
+const getContext = () => {
+  const { location, navigator, innerHeight, innerWidth } = window;
+  const { hash, host, hostname, href, origin, pathname, port, protocol, search } = location;
+  const { language, userAgent } = navigator;
+  const { title, referrer } = document;
+
+  return {
+    window: {
+      location: {
+        hash,
+        host,
+        hostname,
+        href,
+        origin,
+        pathname,
+        port,
+        protocol,
+        search,
+      },
+      navigator: { language, userAgent },
+      innerHeight,
+      innerWidth,
+    },
+    document: {
+      title,
+      referrer,
+    },
+  };
+};
+
+const useFrame = (): UseChatType => {
+  const ref = useRef<HTMLIFrameElement>(null);
+  const chat = useChat();
+  const [style, setStyle] = useState<CSSProperties>({});
+
+  useEffect(() => {
+    const handleMessage = (event: MessageEvent): void => {
+      const { current: chatIframe } = ref;
+
+      if (
+        !chat.enabled ||
+        !chatIframe?.contentWindow ||
+        event.source !== chatIframe?.contentWindow
+      ) {
+        return;
+      }
+
+      const { data: message } = event;
+      const context = getContext();
+
+      switch (message.type) {
+        case MESSAGE_READY: {
+          const user = {
```

Comment:
I'm wondering if we should add a layer of abstraction between a user / chat and the actual chat provider used, so that we have a user with `id` and `email`, and then a Drift user with `jwt`. Could be an overkill at this point, but just thinking out loud.

Reply:
Believe me, that's the goal, (see #123339 )